files/generate_ansible_command.py

```diff
@@ -230,30 +230,35 @@ def get_hostname(x):
             # sure there is no funky chars for shell, and no space
             if re.search('^[\\w.-]+$', hostname):
                 # avoid using the ssh stuff on salt bus host
-                h = filter((lambda f: f['name'] == hostname), new)[0]
-                if h['connection'] == 'ssh':
-                    commands_to_run.append("ssh %s -o "
-                                           "PreferredAuthentications=publickey"
-                                           " -o StrictHostKeyChecking=no %s id"
-                                           % (h['ssh_args'], hostname))
+                commands_to_run.append({'cmd': "ansible -m ping -l %s all"
+                                        % hostname,
+                                        'env':
+                                        {"ANSIBLE_HOST_KEY_CHECKING":
+                                         "false"}})
 
         for p in get_playbooks_to_run(args.path):
             if os.path.basename(p) in configuration['run_on_new_host']:
                 playbooks_to_run.add(p)
 
 if update_requirements:
-    commands_to_run.append('sudo /usr/local/bin/update_galaxy.sh')
+    commands_to_run.append({'cmd': 'sudo /usr/local/bin/update_galaxy.sh'})
 
 for p in playbooks_to_run:
     if os.path.exists(p):
-        commands_to_run.append('ansible-playbook -D %s' % p)
+        commands_to_run.append({'cmd': 'ansible-playbook -D %s' % p})
 
 for p in local_playbooks_to_run:
-    commands_to_run.append('sudo /usr/local/bin/ansible_local.py %s' % p)
+    commands_to_run.append({'cmd':
+                            'sudo /usr/local/bin/ansible_local.py %s' % p})
 
 for c in commands_to_run:
     if args.dry_run:
-        print c
+        print c['cmd']
     else:
         syslog.syslog("Running {}".format(c))
-        subprocess.call(shlex.split(c), cwd=args.path)
+        env = os.environ.copy()
+        for k, v in c.get('env', {}).iteritems():
+            env[k] = v
+        subprocess.call(shlex.split(c['cmd']),
```

Comment:
While you're at it, It would be even safer to pass an array instead of constructing the command in a string by yourself.